src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs

```diff
@@ -43,7 +43,7 @@ public static partial class SymbolFinder
                         if (member != null &&
                             member.IsOverride &&
                             member.OverriddenMember() != null &&
-                           SymbolEquivalenceComparer.Instance.Equals(member.OverriddenMember().OriginalDefinition, symbol.OriginalDefinition))
+                            DependentTypeFinder.OriginalSymbolsMatch(member.OverriddenMember().OriginalDefinition, symbol.OriginalDefinition, solution, cancellationToken))
```

Comment:
Notice this check is for members. Is there another place we may want to update for types themselves?

PS: don't have source access currently, so just guessing from the diff.